### ISSUE TITLE
Restrict opening_brace to single-line declarations

### ISSUE THREAD

**Issue by lilyball** [realm/SwiftLint#1921]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

I'd like to be able to configure the opening_brace rule so it only applies to single-line declarations. If I'm splitting a method across multiple lines then it's usually a good idea to put the opening brace on a new line as well, but single-line declarations should still always have the brace on the same line.

**Comment by kballard** [comment]
For example, the following looks weird:

```swift
func foo<T>(_ arg1: T)
    where T: Equatable {
    // bar
}
```

This reads much better when I can put the brace on a new line:

```swift
func foo<T>(_ arg1: T)
    where T: Equatable
{
    // bar
}
```

**Comment by georgejlee** [comment]
I agree, I have some functions with many long arguments so I find it easier to read when I split the arguments across multiple lines and put the opening brace on its own line, like this:

```
    open class func request(
        _ URLRequest: URLRequestConvertible,
        success: (((NSDictionary) -> Void)?),
        failure: (failureCallback?),
        isSuccess: @escaping ((DataResponse<Any>) -> Bool) = WebCom.defaultIsSuccess)
    {
        ...
    }
```

**Comment by exevil** [comment]
+1 here.

**Comment by hartbit** [comment]
+1 here too

**Comment by petereddy** [comment]
+1

I'm hoping to use SwiftLint to enforce just the thing kballard describes. I still want it to be possible to write:
> func something() -> String {
>     return "hi"
> }

But in the case of functions with multi-line parameters/declarations, I'd like the opening brace to be allowed on a new line for clarity.

**Comment by marcelofabri** [comment]
PRs adding this as a configuration are welcome!

**Comment by kubatruhlar** [comment]
This is a good idea. If we go vertically, it's usually better to put the `{` on its own line to optically separate the parameters from the body of a function.

**Comment by kballard** [comment]
This is becoming more of an issue on my team. I'm not willing to re-enable the `opening_brace` rule without the ability to restrict to single-line declarations, but my coworkers are starting to grumble about PRs being submitted that incorrectly put the opening brace on a new line for single-line declarations.

I am not at all familiar with the implementation of SwiftLint, so I can't even estimate how much work it would be to implement this (for example, does the rule have easy access to information about what line the statement started on, or is it going purely off of the token stream?). But I would really like to see it implemented. What can I do to try and push this forward?

**Comment by remembermyhentai** [comment]
+ 1

same for multiline class declarations (e.g. google swift style guide)

This is much more readable
`
class MyClass:
  MySuperclass,
  MyProtocol,
  SomeoneElsesProtocol,
  SomeFrameworkProtocol
{
  // ...
}
`
